### DISCUSSION TITLE
Sandbox with a Data Exchange Layer

### DISCUSSION THREAD

**Discussion by euanmillar** [spdci/.github#11] [General]
It would be great to host a permanent sandbox that all the projects can use to implement & test the APIs.  

As part of the G2PConnect work last year, many of the collaborators set up their tools on [AWS](https://github.com/orgs/G2P-Connect/discussions/6) and I believe those environments still exist although the installation on there of OpenCRVS is out of date.  

Could we re-purpose the G2PConnect sandbox environments and bring them up to date or alternatively begin discussions regarding a new sandbox location? I know that Accessh are in need of a hosted environment they can develop on too.  They have been using our ID4Africa Hackathon environment to date which is not a permanent solution.

I also noticed discussions on G2PConnect [Github](https://github.com/orgs/G2P-Connect/discussions/26) regarding a Data Exchange layer using [X-Road](https://x-road.global/) and perhaps that could be another nice reference implementation example we could install on the sandbox to demonstrate a robust way to link the organisations together.

**Comment by dasunhegoda** [comment]
Hi @euanmillar,

I agree with your suggestion, too. Having a sandbox environment would indeed be beneficial for future projects and for testing out APIs.

If the X-Road setup turns out to be complex, perhaps we could kick things off with an ESB or MQ. This might be simpler to deploy and set up initially.

**Comment by ghost** [comment]
Hi @dasunhegoda @euanmillar 

The api sandbox environment  development is in progress, very soon it will be available to test sample APIs.
